drivers/include/shtc1.h

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2017, RWTH Aachen. All rights reserved.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_shtc1 SHTC1 Temperature and humidity sensor
+ * @ingroup     drivers_sensors
+ * @ingroup     drivers_saul
+ * @brief       Driver for the Sensirion SHTC1 sensor.
+ *
+ * @file
+ * @brief       Device driver interface for the SHTC1 Temperature and humidity sensor
+ *
+ * @author      Steffen Robertz <steffen.robertz@rwth-aachen.de>
+ * @author      Josua Arndt <jarndt@ias.rwth-aachen.de>
+ */
+
+#ifndef SHTC1_H
+#define SHTC1_H
+
+#include <stdint.h>
+#include "saul.h"
+
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   SHTC1 Default Address
+ *
+ */
+#ifndef SHTC1_I2C_ADDRESS
+#define SHTC1_I2C_ADDRESS         (0x70)
+#endif
+
+
+/**
+ * @brief   enumeration to enable or disable the crc-check
+ * @{
+ */
+typedef enum {
+    CRC_DISABLED = 0,   /**< Disable CRC Check */
+    CRC_ENABLED         /**< Enable CRC Check */
+} shtc1_crc_type_t;
+
+/**
+ * @brief   value struct, where all results obtained from the sensor will be saved.
+ * @{
+ */
+typedef struct {
+    int16_t temp;          /**< Temperature in centi °C (2372 = 23.72) after an call to the measurement function. */
+    uint16_t rel_humidity; /**< Relative humidity in centi percent (9922 = 99.22) after an call to the measuerment function. */
+    unsigned int id;       /**< ID read from the sensor, only available after shtc1_id() was called. */
+} shtc1_values_t;
+/** @} */
+
+/**
+ * @brief   settings struct with all relevant parameters
+ * @{
+ */
+typedef struct {
+    i2c_t i2c_dev;              /**< I2C bus descriptor. */
+    uint8_t i2c_addr;           /**< I2C address of the sensor. */
+    shtc1_crc_type_t crc;       /**< crc check enabled or disabled (CRC_ENABLED/CRC_DISABLED). */
```

Comment:
Does it make sense to have this run-time configurable?

Reply:
you mean the crc check?

Reply:
Yes.
I'm just wondering why someone would set this on a per-device basis or be able to change this at run-time.

Making it a compile-time option should be enough imho.

Reply:
I would remove this because I think the crc-check belongs to a complete measurement...